File: tests/dynamic_fixtures/js/fileio_positive.js
// File I/O — positive fixture.
// Vulnerable: reads a file at a user-controlled path without sanitization.
// Entry: readFile(userPath)  Cap: FILE_IO
// Expected verdict: Confirmed (../../../../etc/passwd → "root:" in output)

const fs = require('fs');
const path = require('path');

function readFile(userPath) {
    const filePath = path.join('/var/data', userPath);
    process.stdout.write('__NYX_SINK_HIT__\n');
    try {
        const content = fs.readFileSync(filePath, 'utf8');
        process.stdout.write(content);
    } catch (e) {
        // Silent on error — target file may not exist in all envs
    }
}

module.exports = { readFile };

File: tests/dynamic_fixtures/go/fileio_positive.go
// File I/O — positive fixture.
// Vulnerable: reads file at user-controlled path without sanitization.
// Entry: ReadFile(userPath string)  Cap: FILE_IO
// Expected verdict: Confirmed (../../../../etc/passwd → "root:" in output)

package entry

import (
	"fmt"
	"os"
	"path/filepath"
)

func ReadFile(userPath string) {
	filePath := filepath.Join("/var/data", userPath)
	fmt.Print("__NYX_SINK_HIT__\n")
	data, err := os.ReadFile(filePath)
	if err == nil {
		fmt.Print(string(data))
	}
}
